src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md

```diff
@@ -0,0 +1,222 @@
+
+The following example shows a code snippet that uses the [preact](https://preactjs.com/) JavaScript framework:
+
+```js
+import { createElement } from 'preact';
+
+export default function App() {
+  return (
+    <p class="big">Hello World!</p>
+  )
+}
+```
+
+In the example, notice that the framework has a lot more control over the code you write, and in some cases, the framework even takes control over when to execute your code.
+
+## Why use a library?
+
+Using a JavaScript library can help to avoid unnecessary [code repetition](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). Libraries can abstract away complex logic, such as date manipulation or financial calculations. A library can also help get your [initial product](https://en.wikipedia.org/wiki/Minimum_viable_product) out, rather than having to write all the code from scratch, which can take time.
+
+Some client-side JavaScript libraries help abstract away quirks of the web platform. A library can also serve as a learning tool. For example, if you're unfamiliar with animation easing [functions](https://easings.net/), the source code of a library can teach you how such easings work.
+
+Some libraries are backed by large companies that invest time and money into keeping libraries up to date and secure. Many libraries are accompanied by extensive documentation, which offers you and your team a quick way to familiarize yourself with the library's usage.
+
+Ultimately, using a JavaScript library saves you time.
+
+## Why should you care about library usage?
+
+Technically, you can develop your web application from scratch, but why go to the trouble when you can use free (open source) software, or purchase a solution that, in the long run, can save time and money? There are a large number of JavaScript libraries and frameworks available, each offering a unique approach to solving problems, and each having different characteristics. For example:
+
+-  A library can be written and maintained internally rather than by a third party.
+-  A library can have specific legal licenses that make it suitable or unsuitable for your web application.
+-  A library can be outdated or unmaintained.
+-  A library can simplify a set of complex tasks and save you a lot of time and money.
+-  A library can be widely used in the community, and can be well known amongst developers.
+
+As you might suspect, different characteristics can affect your web application in different ways. Sometimes, the decision is just not that deep, and you can safely swap out a library if you don't like it. However, sometimes a library can have a significant effect on your work and your web application, which suggests a more informed approach could be necessary.
+
+{% Aside %}
```

Comment:
```suggestion
{% Aside 'important' %}
```

Reply:
I feel this rises to the level of an 'important' aside, since it's a key performance consideration.